src/config_db/ska_sdp_config/config.py

```diff
@@ -301,9 +301,10 @@ def is_processing_block_owner(self, pb_id: str) -> bool:
         Check whether this client is owner of the processing block.
 
         :param pb_id: Processing block ID to look up
-        :returns: Whether processing block is owned
+        :returns: Whether processing block exists and is claimed
```

Comment:
Good change. Like this very much!

Reply:
There is some semantic trickiness here - while the configuration database has a hierarchical structure, this is not enforced. So if you delete a processing block, it doesn't automatically clean up everything "below".

I am not entirely sure how we are going to handle this, especially because for instance deployments are not actually "below" in terms of address. So this might just boil down to nominating some controller (in this case the processing controller presumably) to do clear-up jobs. But until we get there, there's no reason not do do such robustness changes. Might cause a bit of extra traffic from the database, but read access is cheap.

Reply:
Clean-up isn't currently in spec, so we'll defer the complexity!